Added Lava-Animation

anaghkanungo7 opened the pull request:
- Lava animation for Hacktober Fest Animation
- Water animation effect with red color
- Part of my learning Advanced CSS Animations and Transformatins
- Idea adapted from YouTube video : https://www.youtube.com/channel/UCbwXnUipZsLfUckBPsC7Jog
- Open to feedback/criticism :)

anaghkanungo7 commented on animations/lava-animation.css:
```diff
@@ -0,0 +1,43 @@
+* {
+	margin: 0;
+	padding: 0;
+	box-sizing: border-box;
+	font-family: 'Casual Contact MF';
+}
+body {
+	display: flex;
+	justify-content: center;
+	align-items: center;
+    min-height: 100vh;
+    background: #000;
+    white-space: nowrap;
+}
```
Thanks for the feedback Let me make the necessary changes right now. I appreciate the constructive criticism! 👍🏼

anaghkanungo7 commented:
I just made changes - Can you check now?
Here's what I changed:
- Added a wrapper container called outer-container.
- Applied the changed body styles to this outer-container so that it is not dependent on the body style.
- Changed the colors of the text stroke to match the fill.
- Changed the text transforms to make the heading fit inside the wrapper container. 
- Changed the font size a little bit.

Note; The text transforms CSS property can be changed to adjust the text inside the container to be more flexible and compatible with other text animations. The CSS width and height property of the container can also be changed.

Please let me know if I need to make any more changes!

NiallEccles commented:
Hey there! 👋 I still don't think it's working correctly on my end. What I can see is just a large black box with the 'Hacktoberfest animations' title in white.

I had a look at your most recent code and there is a few things I should mention!

You don't have the name of the class in your CSS file matching the name of your animation. The JavaScript that makes all this work takes the name of your animation from `animations.js` and uses that as the class name. So they should match. 

Easy fix!

Even when doing that I still don't think the animation works correctly. I have tinkered with it on my end to get something working but I am unsure if it what you had in mind when making the animation.

Here is the 'updated' code I am working with. I have also attached a small screen recording of it. Is this what you meant?

If not, we can fix it!

**Screen recording:**
[https://www.loom.com/share/6049940afe044e3e820f8af7d25ca8ab](https://www.loom.com/share/6049940afe044e3e820f8af7d25ca8ab)

**Code:**
`.lava-animation{    
    margin: 0;
	padding: 0;
	font-family: 'Courier';
    white-space: nowrap;
}

.lava-animation::after{
    content: attr(data-text);
    position: absolute;
    color: transparent;
    left: 50%;
    -webkit-text-stroke: 2px #DB5521;
    animation: wave 4s ease-in-out infinite;
    transform: translate(-50%);
}

.lava-animation h1:nth-child(2) {
	color: #DB5521;
	animation: wave 4s ease-in-out infinite;
} 

@keyframes wave {
	0%,100% 
	{
		clip-path: polygon(0 42%, 13% 48%, 26% 55%, 41% 64%, 56% 65%, 69% 58%, 84% 45%, 100% 38%, 100% 100%, 0% 100%);
	}
	50%
	{
		clip-path: polygon(0 63%, 14% 57%, 25% 50%, 40% 42%, 56% 40%, 71% 44%, 84% 50%, 100% 60%, 100% 100%, 0% 100%);
	}
}
/*Credits to Online Tutorials
https://www.youtube.com/channel/UCbwXnUipZsLfUckBPsC7Jog*/`

anaghkanungo7 commented:
Hi,

Thanks for the comment. I just changed my animation CSS file to fit the description and requirements. It looks like my old animation wasn't compatible with the rules, since it used two heading tags with different styles. So, I changed it up a bit to fit the rules, and added a GIF to achieve what required two heading tags.

 It should work on your end now. 
I apologize for the inconvenience, I am still learning 😄  

Kindly let me know if there's anything else to be done! If not, kindly merge it to the main branch! Thanks.

NiallEccles commented:
Yes! Looks great!

NiallEccles commented:
No inconvenience! We're all learning here! Thank you for your contribution!
